### ISSUE TITLE
async_server.  when is it safe to destroy  the server object?

### ISSUE THREAD

**Issue by microcai** [cpp-netlib/cpp-netlib#484]
when user calls async_server::stop , no notification method has been defined to notify if the server had actually been stopped and hence save to destroy the object.

**Comment by deanberris** [comment]
That's true. It should be easy to add a callback for when the server is stopped.
